airflow/providers/common/sql/hooks/sql.py

```diff
@@ -202,9 +204,28 @@ def placeholder(self):
             )
         return self._placeholder
 
+    @property
+    def connection(self) -> Connection:
+        if self._connection is None:
+            self._connection = self.get_connection(self.get_conn_id())
+        return deepcopy(self._connection)
```

Comment:
Interesting approach. I think I don't see any big problem except that it will be a Connection ORM object that gets copied and I am not sure what happens in this case and whether there is a problem with some memory leaks and references potentially?

Reply:
No idea either, but I just moved that code from the OdbcHook into the DbApiHook, no idea why they did a deepcopy of the connection in the OdbcHook, will be a reason for it...

Reply:
https://github.com/apache/airflow/blame/main/airflow/providers/odbc/hooks/odbc.py

I don't see usage of `deepcopy` in `OdbcHook`
I'd rather keep it for now without `deepcopy` for the reasons that Jarek stated (memory leaks etc.), and only if we understand its necessity - put it back.